packages/@aws-cdk/cdk/lib/util/validation.ts

```diff
@@ -0,0 +1,12 @@
+import { IConstruct } from "../core/construct";
+
+/**
+ * Validate the given construct and throw if there are any errors
+ */
+export function validateAndThrow(construct: IConstruct) {
+  const errors = construct.node.validateTree();
+  if (errors.length > 0) {
+    const errorList = errors.map(e => `[${e.source.node.path}] ${e.message}`).join('\n  ');
```

Comment:
I like using `\t` for indents in messages because it gives users more control over how it renders (it's more accessible).